custom_commands.py

```diff
@@ -289,6 +289,19 @@ def snappy_compression(self, comp):
         return self
 
 
+class ExternalStage(ClauseElement):
+    """External Stage descriptor"""
+    __visit_name__ = "external_stage"
+
+    def __init__(self, name, path='', namespace=''):
+        self.name = name
+        self.path = '/{}'.format(path) if path and not path.startswith('/') else path
+        self.namespace = '{}.' if namespace and not namespace.endswith('.') else namespace
```

Comment:
I think these lines could be problematic, I can see users passing in `None` instead of `''` and then this String format will be incorrectly `/None` and `None.`

Reply:
Yeah I was thinking about that too and settled on default empty strings for now but am totally up for suggestions.

For paths, if passed a string and it doesn't start with a slash, we need to prepend the slash. Vice versa for the namespace and "."

Either way, what's the neat way to document this? README.rst?